File: java/parser/src/main/java/gov/lbl/parser/parser/Simple_expressionVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Logical_expression;
import gov.lbl.parser.domain.Simple_expression;

import static java.util.stream.Collectors.toList;

public class Simple_expressionVisitor  extends modelicaBaseVisitor<Simple_expression> {
    @Override
    public Simple_expression visitSimple_expression(modelicaParser.Simple_expressionContext ctx) {
        Logical_expressionVisitor logical_expressionVisitor = new Logical_expressionVisitor();
        List<Logical_expression> logical_expressions = ctx.logical_expression() == null ? null : ctx.logical_expression()
                                                                                                        .stream()
                                                                                                        .map(log_expr -> log_expr.accept(logical_expressionVisitor))
                                                                                                        .collect(toList());

        Logical_expression logical_expression1 = logical_expressions.get(0);
        Logical_expression logical_expression2 = null;
        Logical_expression logical_expression3 = null;
        if (logical_expressions.size() > 1) {
            logical_expression2 = logical_expressions.get(1);
        }
        if (logical_expressions.size() > 2) {
            logical_expression3 = logical_expressions.get(2);
        }
        return new Simple_expression(logical_expression1, logical_expression2, logical_expression3);
    }
}

File: jsParser/parser/Simple_expressionVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Simple_expression = require('../domain/Simple_expression')

const Logical_expressionVisitor = require('./Logical_expressionVisitor')

class Simple_expressionVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitSimple_expression (ctx) {
    const logical_expressions = []
    let logical_expression1 = null
    let logical_expression2 = null
    let logical_expression3 = null

    if (ctx.logical_expression()) {
      const logical_expressionVisitor = new Logical_expressionVisitor.Logical_expressionVisitor()
      ctx.logical_expression().forEach(expr => {
        logical_expressions.push(logical_expressionVisitor.visitLogical_expression(expr))
      })
    }

    if (logical_expressions.length > 0) {
      logical_expression1 = logical_expressions[0]

      if (logical_expressions.length > 1) {
        logical_expression2 = logical_expressions[1]
      }
      if (logical_expressions.length > 2) {
        logical_expression3 = logical_expressions[2]
      }
    }

    return new Simple_expression.Simple_expression(logical_expression1, logical_expression2, logical_expression3)
  }
};

Simple_expressionVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitSimple_expression = this.visitSimple_expression
exports.Simple_expressionVisitor = Simple_expressionVisitor
